ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/StateRef.java

```diff
@@ -74,21 +70,17 @@ public int getIndex() {
      */
     public static StateRef parse(@NotNull final String value, DigestService digestService) {
         try {
-            if (!value.contains(DELIMITER)) {
-                throw new IllegalArgumentException(
-                        MessageFormat.format("Failed to parse a StateRef from the specified value. The value: {0} doesn't contain expected delimiter (:)", value)
-                        );
-            }
-            if (value.length() < MINIMUM_HASH_LENGTH) {
+            final int lastIndexOfDelimiter = value.lastIndexOf(DELIMITER);
+            if (lastIndexOfDelimiter == -1) {
                 throw new IllegalArgumentException(
-                        MessageFormat.format("Failed to parse a StateRef from the specified value. The length of the value: {0} doesn't meet the required SHA-256D algorithm length: 73", value.length())
+                        MessageFormat.format("Failed to parse a StateRef from the specified value. The delimiter (:) is missing: {0}.", value)
```

Comment:
I would probably use the `DELIMITER` constant in this message, as it guards against changing the character in future.
```suggestion
                        MessageFormat.format("Failed to parse a StateRef from the specified value. The delimiter ($DELIMITER) is missing: {0}.", value)
```

Reply:
(Just realised the suggestion is Kotlin, not Java, but you get the idea :))